### PR TITLE
Post with redux:

### PR DESCRIPTION
**Pull request by vaishnavipaliya11**
User can login and add their post .
User can edit their post.
User can delete their post.
image
image

### CODE REVIEW

File: src/features/postSlice.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      state.id = action.payload;
+    },
+  },
+  extraReducers: {
+    [getPost.pending]: (state) => {
+      state.status = "loading";
+      state.error = null;
+    },
+    [getPost.fulfilled]: (state, { payload }) => {
+      console.log(payload);
```
Comment (SK02K1) [inline src/features/postSlice.js]: remove this before merging

File: src/Components/Post/Post.jsx [hunk 2]
```diff
@@ -24,8 +24,8 @@ const Post = ({ post }) => {
             className="avatar-img"
             src="https://resize.indiatvnews.com/en/resize/newbucket/715_-/2016/05/shaktiman-1462557537.jpg"
           />
-          <h3 className="user-name">{username}</h3>
-          <p className="user-id">@{username}</p>
+          <h3 className="user-name">{post.username}</h3>
+          <p className="user-id">@{post.username}</p>
```
Comment (dev-enforced) [inline src/Components/Post/Post.jsx:28]: Can use optional chaining process for checking whether the post object is coming clearly or not and no error comes up while rendering it!!
```
{post?.username}
```

File: src/features/postSlice.js (hunk 1 above)
Comment (Pratik1005) [inline src/features/postSlice.js]: remove console log if not needed

File: src/Components/modal/editModal.js [hunk 3]
```diff
@@ -26,16 +31,13 @@ export const EdiPostModal = ({ isOpen, onClose }) => {
           <ModalHeader>Edit Tweet</ModalHeader>
           <ModalCloseButton />
           <ModalBody>
-            <Input onChange={(e) => setEditPost(e.target.value)} />
+            <Input onChange={(e) => setEdittedPost(e.target.value)} />
```
Comment (Pratik1005) [inline src/Components/modal/editModal.js:34]: When we click on edit , the input is not showing the post its blank , I think you missed to add value attribute which should hold the post we are editing.

**Review by dev-enforced — APPROVED** [review]
The code structure looks really good !! And the live preview functionalities also are working correctly as mentioned in the PR comment!!
Had a few suggestions:
1) While handling errors in try catch block you can try using console.error for showing the errors instead of console.log which helps out while debugging and understanding
2) When I first opened the live preview and clicked on the Create Post modal the create button was disabled which confused me for a while as to why that happened but then got to realise that I wasn't logged in so that was a clever way of indicating users to log in and then start creating post!! But you can indicate in modal as a small msg for better clarity itself that user needs to sign in in order to post something
3) While editing content of the existing posts it isn't getting updated even after entering content in the edit modal part of it
image
3) When the edit post modal comes up should it not have the content of the post that is about to be edited before we start modifying it? 
Awesome work @vaishnavipaliya11 😃 these were my suggestions and general queries I had related to the working Learnt a lot about redux code splitting!!

**Review by Pratik1005 — APPROVED** [review]
I have added couple of suggestion, rest look good

**Review by SK02K1 — APPROVED** [review]

**Review by deepak29-git — APPROVED** [review]
code looks good and variable name is also good

**Pull request merged**